File: C++/24.cpp
#include<iostream.h>
int main()
{
    int n, n1, num = 0;
    cout << " Enter a positive integer :  ";
    cin >> n;

    n1 = n;
    while (n != 0)
    {
        n /= 10; 
        num++; 
    }
    cout << "\n\nThe number of digits in the entered number: " << n1 << " is " << num;
    cout << "\n\n\n";

    return 0;
}

File: Java/24.java
import java.util.Scanner;

class NumberOfDigitsInInteger {
    public static void main(String[] args) {
        int count = 0, num;
        Scanner a = new Scanner(System.in);
        System.out.println("Enter integer :");
        num = a.nextInt();
        a.close();
        while (num != 0) {
            // num = num/10
            num /= 10;
            ++count;
        }
        System.out.println("Number of digits: " + count);
    }
}